lua/lean/infoview.lua

```diff
@@ -224,8 +243,10 @@ end
 
 --- Open an infoview for the current buffer if it isn't already open.
 function infoview.maybe_autoopen()
-  local id = get_id()
-  if not infoview._by_id[id] then infoview._by_id[id] = Infoview:new(id, options.width, options.autoopen) end
+  local tabpage = vim.api.nvim_win_get_tabpage(0)
```

Comment:
Why move towards storing things by tabpage when there will be multiple views on one tabpage?

Reply:
Assume you're referring to per-window infoviews? This is where we'd initialize separate infoviews for both the tab and the window.

Reply:
`get_id` semantically before meant, to an internal maintainer (i.e. us), "this is the index to store or retrieve an infoview from". Now this function reaches across that boundary and touches arbitrary state (which other functions rely on), so now that implementation detail is scattered again in multiple places, when I was hoping we'd go in the other direction and eventually remove `get_id` entirely. But it's fine, for now we need to get past continually refactoring this file without functional changes -- let's get to where there's functionality being changed and then I'll worry about it.